### ISSUE TITLE
matchparent plugin shouldn't use the match ID 3

### ISSUE THREAD

**Issue by nicoe** [neovim/neovim#17157]
### Neovim version (nvim -v)

0.6.1

### Vim (not Nvim) behaves the same?

yes,  8.2.3995 

### Operating system/version

Debian Unstable

### Terminal name/version

foot 1.10.3

### $TERM environment variable

foot

### Installation

Debian package

### How to reproduce the issue

nvim clean
:3match Error /\s\+$/
type `(foo)`

When the closing parenthesis is typed, an error is raised.

### Expected behavior

No error.

I think what the matchparen plugin should do (but I am not competent in vim scripting enough to do it) is use -1 as ID the first time it's called, store this ID and then reuse it afterwards. It's just a suggestion :smile: (as I said I don't know much about vim scripting and I am a bit puzzled by the fact that this hasn't been reported before (I didn't search vim bugs)).

All of this happens here: https://github.com/neovim/neovim/blob/c977d8b43cd6ecf7ad756f9b064eadea79fbd604/runtime/plugin/matchparen.vim#L188

### Actual behavior

```
Error detected while processing function <SNR>6_Highlight_Matching_Pair:                                                                                                                                  
line  148:
E801: ID already taken: 3
```

**Comment by chrisbra** [comment]
Quoting: https://vimhelp.org/pattern.txt.html#%3A3match

>  The ":3match" command is used by the matchparen plugin.  You are suggested to use ":match" for manual matching and  ":2match" for another plugin.

**Comment by nicoe** [comment]
I should have read the documentation more extensively :facepalm: (I ended up using :2match indeed).
It's a bit of a pitty though.

**Comment by chrisbra** [comment]
I would rather suggest to use the `matchadd()` functionality instead. It's much more flexible.

**Comment by zeertzjq** [comment]
Fixed by #25734
